OSDOCS-3209 - Adding OSD wizard updates (#41560)
openshift/openshift-docs: modules/osd-create-cluster-ccs.adoc

// Module included in the following assemblies:
//
// * osd_cluster_create/creating-an-aws-cluster.adoc
// * osd_cluster_create/creating-a-gcp-cluster.adoc

ifeval::["{context}" == "osd-creating-a-cluster-on-aws"]
:osd-on-aws:
endif::[]
ifeval::["{context}" == "osd-creating-a-cluster-on-gcp"]
:osd-on-gcp:
endif::[]

:_content-type: PROCEDURE
ifdef::osd-on-aws[]
[id="osd-create-aws-cluster-ccs_{context}"]
= Creating a cluster on AWS with CCS
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
[id="osd-create-gcp-cluster-ccs_{context}"]
= Creating a cluster on GCP with CCS
endif::osd-on-gcp[]

By using the Customer Cloud Subscription (CCS) billing model, you can create an {product-title} cluster in an existing 
ifdef::osd-on-aws[]
{AWS} 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
{GCP} 
endif::osd-on-gcp[]
account that you own.

> You must meet several prerequisites if you use the CCS model to deploy and manage {product-title} into your AWS account.

Reviewer: +1 AWS shouldn't be here

> You must meet several prerequisites if you use the CCS model to deploy and manage {product-title} into your AWS account.

Author: Good catch! I have subsequently added the required conditionals.

.Prerequisites

ifdef::osd-on-aws[]
* You have configured your AWS account for use with {product-title}.
* You have not deployed any services in your AWS account.
* You have configured the AWS account quotas and limits that are required to support the desired cluster size.
* You have an `osdCcsAdmin` AWS Identity and Access Management (IAM) user with the `AdministratorAccess` policy attached.

> * You have set up a service control policy (SCP) in your AWS organization.

Reviewer: UI shows `An Organization Service Control Policy (SCP) is set up according to the requirements for customer cloud subscriptions.` better to keep consistent. And if there is a doc ULR about SCP policy requirement will be better.

Author: In the AWS documentation, this is referred to as a "service control policy". I prefer to add "in your organization" rather than use "organizational" in the docs. I think it is more explicit.

That's a great idea about adding a link. Because we cannot include cross-references in modules, I have subsequently updated the prerequisite to the following:

"* You have set up a service control policy (SCP) in your AWS organization. For information, see Minimum required service control policy (SCP)."

I have also added an additional resources link to that section at the end of the assembly, so that the string and a link can be found on the same page.

* Consider having *Business Support* or higher from AWS.
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
* You have configured your GCP account for use with {product-title}.
* You have configured the GCP account quotas and limits that are required to support the desired cluster size.
* You have created a GCP project.
+
[NOTE]
====
The project name must be 10 characters or less.
====
* You have enabled the Google Cloud Resource Manager API in your GCP project. For more information about enabling APIs for your project, see link:https://cloud.google.com/endpoints/docs/openapi/enable-api[the Google Cloud documentation].
* You have an IAM service account in GCP called `osd-ccs-admin` with the following roles attached:
  ** DNS Administrator

>   ** Organization Policy Viewer Owner

Reviewer: The prerequisite is different with UI, would you please double confirm?

Author: Following on from a discussion in Slack with our Engineering team, a review of the details in https://issues.redhat.com/browse/OSDOCS-3194 and my own tests to deploy OSD on GCP with CCS, I have now updated these prerequisites to match what is in the UI.

  ** Project IAM Admin
  ** Service Management Administrator
  ** Service Usage Admin
  ** Storage Admin
* You have created a key for your `osd-ccs-admin` GCP service account and exported it to a file named `osServiceAccount.json`.
+
[NOTE]
====
For more information about creating a key for your GCP service account and exporting it to a JSON file,  see link:https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys[Creating service account keys] in the Google Cloud documentation.
====
* Consider having *Production Support* or higher from GCP.
* To prevent potential conflicts, consider having no other resources provisioned in the project prior to installing {product-title}.
endif::osd-on-gcp[]

.Procedure

> . Log in to {console-redhat-com} and click *Create Cluster*.

Reviewer: On UI it's "Create cluster" rather than "Create Cluster"

Author: Thank you for highlighting this. Resolved.

> . Log in to {console-redhat-com} and click *Create Cluster*.

Reviewer: It's `Create cluster` rather than `Create Cluster` on the UI.

> . In the *Cloud* tab, click *Create Cluster* in the *Red Hat OpenShift Dedicated* row.

Reviewer: On UI it's "Create cluster" rather than "Create Cluster"

Author: Resolved.

> . In the *Billing model* dialog, configure the subscription type and infrastructure type:

Reviewer 2: I suggest 'screen' instead of dialog since it's one screen in a wizard, instead of a box that pops up. (That's what the OCM docs use anyway)

Reviewer 3: to add to this, nothing on that page indicates a billing model.  Maybe this should be called "infrastructure provider"?

Author: This is a great suggestion, thanks! Resolved.

> .. Select a subscription type.

Reviewer 2: This could be a good place to link to the docs about subscriptions in the OCM docs, as once you select whether your cluster is On-Demand or Annual, you can't change it later: https://access.redhat.com/documentation/en-us/openshift_cluster_manager/2022/html-single/managing_clusters/index#subscribing-osd-cluster_assembly-cluster-subscriptions

> .. Select the *Customer cloud subscription* infrastructure type to deploy {product-title} in an existing cloud provider account that you own.

Reviewer 2: Customer Cloud Subscription should be capitalized - I raised this in the OSD wizard review Jira with Dave :)

Reviewer: +1

Author: Resolved.

.. Click *Next*.

ifdef::osd-on-aws[]
. Select *Run on Amazon Web Services* 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
. Select *Run on Google Cloud Platform* 
endif::osd-on-gcp[]

> and click *Next*.

Reviewer: `Next` Here is extra step which not required.

Author: Good catch! Resolved.

. Click *Prerequisites* to review the prerequisites for installing {product-title} on 
ifdef::osd-on-aws[]
AWS 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
GCP 
endif::osd-on-gcp[]
with CCS.

ifdef::osd-on-aws[]
. Provide your AWS account details:
.. Enter your *AWS account ID*.
.. Enter your *AWS access key ID* and *AWS secret access key* for your AWS IAM user account.
+
[NOTE]
====

> Revoking these credentials in AWS will result in a loss of access to any cluster created with these credentials.

Author: Thanks for highlighting this. Resolved.

====
.. Optional: You can select *Bypass AWS Service Control Policy (SCP) checks* to disable the SCP checks.
+
[NOTE]
====

> Some AWS SCPs will cause the installation to fail, even if you have the required permissions. Disabling the SCP checks allows an installation to proceed. The SCP will still be enforced even if the checks are bypassed.

Author: Thanks for the suggestion. Resolved.

====
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
. Provide your GCP service account private key in JSON format. You can either click *Browse* to locate and attach a JSON file or add the details in the *Service account JSON* field.
endif::osd-on-gcp[]

. *Validate* your cloud provider account and then click *Next*.

. On the *Cluster details* page, provide a name for your cluster and specify the cluster details:
.. Add a *Cluster name*.
.. Select a cluster version from the *Version* drop-down menu.
.. Select a cloud provider region from the *Region* drop-down menu.
.. Select a *Single zone* or *Multi-zone* configuration.

> .. Leave *Enable user workload monitoring* selected to monitor your own projects in isolation from Red Hat Site Reliability Engineer (SRE) platform metrics. The option is enabled by default.

Author: Resolved.

> .. Optional: Select *Enable additional etcd encryption* if you require etcd key value encryption. With this option, the etcd key values are encrypted, but not the keys. The option is in addition to the control plane storage encryption that encrypts the etcd volumes in {product-title} clusters by default.

Author: Resolved.

+
[NOTE]
====
By enabling etcd encryption for the key values in etcd, you will incur a performance overhead of approximately 20%. The overhead is a result of introducing this second layer of encryption, in addition to the default control plane storage encryption that encrypts the etcd volumes. Consider enabling etcd encryption only if you specifically require it for your use case.
====
.. Optional: Select *Encrypt persistent volumes with customer keys* if you want to provide your own 
ifdef::osd-on-aws[]

> AWS Key Management Service (KMS) key Amanzon Resource Name (ARN). 

Reviewer 4: ```suggestion
AWS Key Management Service (KMS) key Amazon Resource Name (ARN). 
```

Author: Good catch, thank you! Resolved.

endif::osd-on-aws[]
ifdef::osd-on-gcp[]
encryption keys through the Google Cloud Key Management Service. 
endif::osd-on-gcp[]
The key is used for encryption of persistent volumes in your cluster.
.. Click *Next*.

> . On the *Default machine pool* page, select a *Compute node instance type* and a *Compute node count*. The number and types of nodes that are available depend on your {product-title} subscription. If you are using multiple availability zones, the compute node count is per zone.

Reviewer: UI shows `Worker node instance type` rather than Compute node instance type

Author: I have suggested that the UI is updated to use "compute node" instead of "worker node" in https://issues.redhat.com/browse/HAC-451. I will keep this as is in the docs in preparation for the UI update.

+
[NOTE]
====
After your cluster is created, you can change the number of compute nodes in your cluster, but you cannot change the compute node instance type. The number and types of nodes available to you depend on your {product-title} subscription.
====

> . Optional: Expand *Edit node labels* to add labels to your nodes. Click *Add label* to add more node labels.

Reviewer: Lack of "Click Next button" step to go to next page

Author: Thank you for highlighting this. Resolved.

. In the *Cluster privacy* section, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster.
+
[IMPORTANT]
====
If you are using private API endpoints, you cannot access your cluster until you update the network settings in your cloud provider account.
====

ifdef::osd-on-aws[]
. Optional: To install the cluster in an existing AWS Virtual Private Cloud (VPC):
.. Select *Install into an existing VPC*.

> .. Select *Use a PrivateLink* if you want to enable connections to the cluster by Red Hat Site Reliability Engineering (SRE) using only AWS PrivateLink endpoints instead of public endpoints. This option cannot be changed after a cluster is created.

Author: Thank you for highlighting this. I have updated the step as follows:

".. If you opted to use private API endpoints and are installing into an existing VPC, you can select *Use a PrivateLink* to enable connections to the cluster by Red Hat Site Reliability Engineering (SRE) using only AWS PrivateLink endpoints. This option cannot be changed after a cluster is created."

endif::osd-on-aws[]
ifdef::osd-on-gcp[]
. Optional: Select *Install into an existing VPC* to install the cluster in an existing GCP Virtual Private Cloud (VPC).
endif::osd-on-gcp[]

. Click *Next*.

. If you opted to install the cluster in an existing 
ifdef::osd-on-aws[]
AWS 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
GCP 
endif::osd-on-gcp[]
VPC, provide your *Virtual Private Cloud (VPC) subnet settings*.
ifdef::osd-on-aws[]
+
[NOTE]
====
You must ensure that your VPC is configured with a public and a private subnet for each availability zone that you want the cluster installed into.
====
endif::osd-on-aws[]

> . In the *CIDR ranges* dialog, configure custom classless inter-domain routing (CIDR) ranges or use the defaults that are provided.

Reviewer: `configure custom classless inter-domain routing (CIDR) ranges or use the defaults that are provided` maybe it's not `that` , it should be `those`?

Author: I think "use the defaults that are provided" is valid, so I will keep this line as it is.

+
[IMPORTANT]
====

> CIDR configurations cannot be changed later. Confirm your selections with your network administrator before proceeding.

Reviewer: It's better to also clarify that the machine CIDR should match the VPC subnets once set

====

. On the *Cluster update strategy* page, configure your update preferences:

> .. Choose a cluster update method. By selecting *Individual updates*, you can schedule each update individually. If you select *Recurring updates*, your cluster is updated on your preferred day and start time, when updates are available. *Individual updates* is the default option.

Author: Thank you for this suggestion. I have now implemented this and it scans well!

+
[NOTE]
====

> You can review the end-of-life dates in the update life cycle documentation for {product-title}. For more information, see _{product-title} update life cycle_.

Reviewer 4: ISG would like "lifecycle" to be one word but if it's already two words elsewhere in OSD docs then this is fine as is.

====
+
.. If you opt for recurring updates, select a preferred day of the week and upgrade start time in UTC from the drop-down menus.

> .. Optional: You can set a grace period for *Node Draining* during cluster upgrades. A *1 hour* grace period is set by default.

Reviewer: It's `Node draining` on UI rather than `Node Draining`

Author: Thank you for highlighting this. Resolved.

.. Click *Next*.
+
[WARNING]
====

> High and Critical security concerns (CVEs) will be patched automatically within 48 hours, regardless of your chosen update strategy.

Reviewer 4: ```suggestion
High and Critical security concerns (CVEs) are patched automatically within 48 hours, regardless of your chosen update strategy.
```

Author: Resolved.
